core/utils/xml.ts

```diff
@@ -14,40 +14,69 @@
 import * as goog from '../../closure/goog/goog.js';
 goog.declareModuleId('Blockly.utils.xml');
 
+import * as deprecation from './deprecation.js';
+
 
 /**
- * Namespace for Blockly's XML.
+ * Injected dependencies.  By default these are just (and have the
+ * same types as) the corresponding DOM Window properties, but the
+ * Node.js wrapper for Blockly (see scripts/package/node/core.js)
+ * calls injectDependencies to supply implementations from the jsdom
+ * package instead.
+ */
+let {document, DOMParser, XMLSerializer} = globalThis;
+
+/**
+ * Inject dependencies.  Used by the Node.js wrapper for Blockly (see
+ * scripts/package/node/core.js) to supply implementations from the
+ * jsdom package instead.  While they may be set individually, it is
+ * normally the case that all three should be supplied and sourced
+ * from the same JSDOM instance.
```

Comment:
Isn't setting thing from different JSDOM instances what caused us trouble in the first place? If so I think this inline documentation should be slightly more opinionated :P

Reply:
No, setting things using values _generated by separate copies of the `jsdom` package_ caused the trouble in the first place.  It's actually weird-but-fine to use separate _instances_ of the `JSDOM` class, provided they all came from the same (`===`) constructor.

But you have a point that the documentation could be more opinionated.